fedimint-client/src/lib.rs

```diff
@@ -1469,19 +1469,12 @@ async fn get_config(
     db: &Database,
     config_source: Option<ConfigSource>,
 ) -> anyhow::Result<ClientConfig> {
-    let mut dbtx = db.begin_transaction().await;
-    let config_res = match dbtx
-        .find_by_prefix(&ClientConfigKeyPrefix)
-        .await
-        .next()
-        .await
-    {
-        Some((_, config)) => {
-            // TODO: Enable after <https://github.com/fedimint/fedimint/pull/2855>
-            // assert!(
-            //     config_source.is_none(),
-            //     "Alternative config source provided but config was found in DB"
-            // );
+    let config_res = match get_config_from_db(db).await {
+        Some(config) => {
+            assert!(
+                config_source.is_none(),
+                "Alternative config source provided but config was found in DB"
+            );
```

Comment:
Can we make this an [`ensure!`](https://docs.rs/anyhow/1.0.73/anyhow/macro.ensure.html) please? It feels way too easy to get wrong and we don't want to needlessly blow up the whole application. It's a close call here tbf.